lib/oauther.ex

```diff
@@ -72,19 +72,25 @@ defmodule OAuther do
   end
 
   defp base_string(verb, url, params) do
-    [verb, URI.parse(url), params]
-    |> Stream.map(&normalize/1)
-    |> Enum.map_join("&", &percent_encode/1)
+    {base_url, q_params} = parse_url(url)
+    [String.upcase(verb), base_url, normalize_params(params, q_params)]
+      |> Enum.map_join("&", &percent_encode/1)
+  end
+
+  defp parse_url(url) do
+    uri = URI.parse(url)
+    {"#{uri.scheme}://#{uri.host}#{uri.path}", parse_query_params(uri) }
```

Comment:
I think we could use `get_and_update` way here: get `params` from `URI` and set the to `nil`.
Sadly, `Map.get_and_update/3` is only available in Elixir v1.1, thus we need to reimplement it here.